### PR TITLE
rename FleetCluster to KubeConfigSecretRef 

### PR DESCRIPTION
**Pull request by Xieql**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

current `FleetCluster`  struct is just for render

and we have already defined `fleetCluster `  here

https://github.com/kurator-dev/kurator/blob/2871c99124b0f27a3fcdd9542fb41db0b5cf1d89/pkg/fleet-manager/fleet_cluster.go#L34-L38

and we are going to make it public for #481 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/fleet-manager/fleet_plugin_backup.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ func (f *FleetManager) reconcileBackupPlugin(ctx context.Context, fleet *v1alpha
 	// Iterating through each fleet cluster to generate and apply Velero helm configurations.
 	for key, cluster := range fleetClusters {
 		// generate Velero helm config for each fleet cluster
-		b, err := plugin.RenderVelero(f.Manifests, fleetNN, fleetOwnerRef, plugin.FleetCluster{
+		b, err := plugin.RenderVelero(f.Manifests, fleetNN, fleetOwnerRef, plugin.RenderableFleetCluster{
```
Comment (hzxuzhonghu) [inline pkg/fleet-manager/fleet_plugin_backup.go:84]: can we use the FleetCluster, if any field miss, we can add it

Comment (Xieql) [reply pkg/fleet-manager/fleet_plugin_backup.go:84]: the origin `FleetCluster`' fileld ' value is comes from two struct `fleetCluster`  and `ClusterKey` ,
so maybe we can not direct add filed in `FleetCluster`, or we need merge those two struct into one struct, which may affect many other code

https://github.com/kurator-dev/kurator/blob/2871c99124b0f27a3fcdd9542fb41db0b5cf1d89/pkg/fleet-manager/fleet_cluster.go#L34-L43

this is the most used usage:

https://github.com/kurator-dev/kurator/blob/83fdec03d1ed0b5f01c6e3d5ed5e47f038d4f5f6/pkg/fleet-manager/fleet_plugin_distributedstorage.go#L78-L81

 	FleetCluster.Name:       ClusterKey.Name, 
 	FleetCluster.SecretName: fleetCluster.Secret, 
 	FleetCluster.SecretKey:  fleetCluster.SecretKey,

**Comment by Xieql** [comment]
> Looked into the code closely, and the FleetCluster represents the kubeconfig secret ref.
> 
> We only need the secret name and key which kubeconfg is stored. Even the cluster name is not needed.
> 
> So we can rename it to `KubeConfigSecretRef` or `SecretRef` or something

o'kok

File: pkg/fleet-manager/plugin/render.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ func (plugin FleetPluginConfig) StorageNamespace() string {
 	return plugin.Fleet.Namespace
 }
 
-type FleetCluster struct {
+type KubeConfigSecretRef struct {
 	Name       string
```
Comment (hzxuzhonghu) [inline pkg/fleet-manager/plugin/render.go:59]: nit: The Name can be removed

**Review by hzxuzhonghu — APPROVED** [review]

File: pkg/fleet-manager/plugin/render.go (hunk 2 above)
Comment (Xieql) [reply pkg/fleet-manager/plugin/render.go:59]: Currently, it is used solely for rendering and requires the `name` parameter. 

If we remove `name`, we will need to refactor many corresponding functions

However, I think that refactoring them would not be wise.

such as 
https://github.com/kurator-dev/kurator/blob/5de151793e19756831fddc52b986f6b7309f30b5/pkg/fleet-manager/plugin/render.go#L30-L46

Comment (hzxuzhonghu) [reply pkg/fleet-manager/plugin/render.go:59]: I did see that, it is unused

Comment (Xieql) [reply pkg/fleet-manager/plugin/render.go]: you are right, it is unused

**Comment by Xieql** [comment]
If the `name` field is removed, the test fails with the message 'template: fleet plugin template:4:13: executing "fleet plugin template" at <$.ResourceName>: can't evaluate field ResourceName in type plugin.FleetPluginConfig'. 

Therefore, it might be best to retain the field, as there seems to be no effective method to address this issue

**Review by hzxuzhonghu — COMMENTED** [review]
/lgtm

**Pull request merged**